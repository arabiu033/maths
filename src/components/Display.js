import PropTypes from 'prop-types';
import React from 'react';

class Display extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { obj } = this.props;
    return (
      <div id="screen">
        <span>
          { `${obj.total ? obj.total : ''} 
          ${obj.operation ? obj.operation : ''} 
          ${obj.next ? obj.next : ''}` }
        </span>
      </div>
    );
  }
}

Display.propTypes = {
  obj: PropTypes.objectOf(PropTypes.string).isRequired,
};

export default Display;
