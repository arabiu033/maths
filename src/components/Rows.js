import PropTypes from 'prop-types';
import React from 'react';

class Rows extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { row } = this.props;
    const buttons = [];
    for (let i = 0; i < row.length; i += 1) {
      buttons.push(<button type="button" id="" className="">{ row[i] }</button>);
    }

    return buttons;
  }
}

Rows.propTypes = {
  row: PropTypes.string.isRequired,
};

export default Rows;
