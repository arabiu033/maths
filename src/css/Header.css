* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  text-decoration: none;
}

.container {
  padding: 1rem 2rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 5rem;
  flex-shrink: 1;
}

ul,
li {
  list-style: none;
}

ul {
  display: flex;
  justify-content: space-around;
  width: 30%;
}

li {
  padding-bottom: 1px;
  padding-left: 4rem;
  border-left: 1px solid skyblue;
  font-size: 1.3rem;
}

li span {
  border-bottom: 1px solid skyblue;
  padding-bottom: 1px;
}

a {
  color: skyblue;
}

li:first-child {
  padding: 0;
  border: none;
}

h1 {
  font-size: 4.5rem;
  font-weight: normal;
  letter-spacing: 2px;
}
