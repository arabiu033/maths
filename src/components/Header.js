import React from 'react';
import { NavLink } from 'react-router-dom';
import '../css/Header.css';

const Header = () => (
  <div className="container">
    <h1>Math Magicians</h1>
    <ul>
      <li><span><NavLink to="/">Home</NavLink></span></li>
      <li><span><NavLink to="/calculator">Calculator</NavLink></span></li>
      <li><span><NavLink to="/quote">Quote</NavLink></span></li>
    </ul>
  </div>
);

export default Header;
