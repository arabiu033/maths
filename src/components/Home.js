import React from 'react';
import '../css/home.css';

const Home = () => (
  <div className="home">
    <h2>
      Welcome to our page!
    </h2>
    <div>
      <p>
        Lorem ipsum dolor sit amet, at habemus suscipit comprehensam usu.
        Deserunt scribentur id eam, ex persecuti neglegentur mea, 
        his tantas urbanitas id. Usu oratio antiopam temporibus eu. Sed
        an legimus repudiare, probo civibus his ex.Everti recteque et sea,
        no cetero menandri mei. Ut vel summo omittantur. At inciderint temporibus has.
        Facilis salutatus complectitur ex eam, ex aeque indoctum pro,
        nisl quas nullam mei an. Duo te sensibus mandamus, ei his vidit populo.
      </p>
      <p>
        Ad vim iisque epicuri evertitur. Ne quo doctus detracto,
        eam apeirian expetenda ex, cum dicunt intellegebat ut.
        Duo no signiferumque mediocritatem, id legendos tincidunt pro,
        dicant senserit usu no. Ut oratio cotidieque mel. Cum an zril impedit repudiare,
        id per eius delenit perpetua. Illud mnesarchum has ut, eu vis quem
        voluptatum deterruisset, eos erant expetenda ne. Dicit option
        fabellas mea id. Ea agam regione quo, soluta molestiae posidonium no nam.
        Eu ius tantas appareat. Ex cum aliquip euismod legendos, feugiat consulatu nec ea.
        Et summo evertitur est, sed ne natum aperiam definiebas, dolorem corrumpit ius id.
        Ad ius detracto philosophia.
      </p>
    </div>
  </div>
);

export default Home;
