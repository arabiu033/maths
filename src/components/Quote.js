import React from 'react';
import '../css/quote.css';

const Quote = () => (
  <div className="quote">
    <h3>
      Mathematics is not about numbers, equations, computations,
      <br />
      or algorithms: it is about understanding. -- William Paul Thurston
    </h3>
  </div>
);

export default Quote;
