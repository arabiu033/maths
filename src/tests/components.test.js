import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Display from '../components/Display';
import Home from '../components/Home';
import Header from '../components/Header';
import Rows from '../components/Rows';
import Quote from '../components/Quote';
import Calc from '../components/Calculator';

it('render correctly Display component', () => {
  const obj = {
    total: null,
    next: null,
    operation: null,
  };
  const display = renderer.create(<Display obj={obj} />).toJSON();
  expect(display).toMatchSnapshot();
});

it('render correctly Header component', () => {
  const header = renderer.create(<BrowserRouter><Header /></BrowserRouter>).toJSON();
  expect(header).toMatchSnapshot();
});

it('render correctly Quote component', () => {
  const quote = renderer.create(<Quote />).toJSON();
  expect(quote).toMatchSnapshot();
});

it('render correctly Home component', () => {
  const home = renderer.create(<Home />).toJSON();
  expect(home).toMatchSnapshot();
});

it('render correctly Rows component', () => {
  const row = renderer.create(<Rows row={['1', '2', '3']} setObj={() => {}} />).toJSON();
  expect(row).toMatchSnapshot();
});

it('render correctly Calc component', () => {
  const calc = renderer.create(<Calc />).toJSON();
  expect(calc).toMatchSnapshot();
});
