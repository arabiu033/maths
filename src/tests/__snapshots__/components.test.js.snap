// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly Calc component 1`] = `
<div
  className="calc"
>
  <h1>
    Lets do some math!
  </h1>
  <main>
    <div
      id="screen"
    >
      <span>
         
         
        
      </span>
    </div>
    <div>
      <div
        className="rows"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="rows"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="rows"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="rows"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="rows"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </main>
</div>
`;

exports[`render correctly Display component 1`] = `
<div
  id="screen"
>
  <span>
     
         
        
  </span>
</div>
`;

exports[`render correctly Header component 1`] = `
<div
  className="container"
>
  <h1>
    Math Magicians
  </h1>
  <ul>
    <li>
      <span>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </span>
    </li>
    <li>
      <span>
        <a
          className=""
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </span>
    </li>
    <li>
      <span>
        <a
          className=""
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </span>
    </li>
  </ul>
</div>
`;

exports[`render correctly Home component 1`] = `
<div
  className="home"
>
  <h2>
    Welcome to our page!
  </h2>
  <div>
    <p>
      Lorem ipsum dolor sit amet, at habemus suscipit comprehensam usu. Deserunt scribentur id eam, ex persecuti neglegentur mea, his tantas urbanitas id. Usu oratio antiopam temporibus eu. Sed an legimus repudiare, probo civibus his ex.Everti recteque et sea, no cetero menandri mei. Ut vel summo omittantur. At inciderint temporibus has. Facilis salutatus complectitur ex eam, ex aeque indoctum pro, nisl quas nullam mei an. Duo te sensibus mandamus, ei his vidit populo.
    </p>
    <p>
      Ad vim iisque epicuri evertitur. Ne quo doctus detracto, eam apeirian expetenda ex, cum dicunt intellegebat ut. Duo no signiferumque mediocritatem, id legendos tincidunt pro, dicant senserit usu no. Ut oratio cotidieque mel. Cum an zril impedit repudiare, id per eius delenit perpetua. Illud mnesarchum has ut, eu vis quem voluptatum deterruisset, eos erant expetenda ne. Dicit option fabellas mea id. Ea agam regione quo, soluta molestiae posidonium no nam. Eu ius tantas appareat. Ex cum aliquip euismod legendos, feugiat consulatu nec ea. Et summo evertitur est, sed ne natum aperiam definiebas, dolorem corrumpit ius id. Ad ius detracto philosophia.
    </p>
  </div>
</div>
`;

exports[`render correctly Quote component 1`] = `
<div
  className="quote"
>
  <h3>
    <span>
      Mathematics is not about numbers, equations, computations,
    </span>
    <br />
    or algorithms: it is about understanding. -- William Paul Thurston
  </h3>
</div>
`;

exports[`render correctly Rows component 1`] = `
Array [
  <button
    onClick={[Function]}
    type="button"
  >
    1
  </button>,
  <button
    onClick={[Function]}
    type="button"
  >
    2
  </button>,
  <button
    onClick={[Function]}
    type="button"
  >
    3
  </button>,
]
`;
